### PR TITLE
Feature/batterystate

### PR DESCRIPTION
**Pull request by athackst**

### CODE REVIEW

File: src/task_behavior_ros/batterystate.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                return NodeStatus(NodeStatus.SUCCESS, "charge above max")
+            else:
+                return NodeStatus(NodeStatus.ACTIVE, "charging")
+        return NodeStatus(NodeStatus.FAIL, "not charging")
+
+
+class ChargeOKMonitor(TopicMonitor):
+
+    """
+    Checks if msg indicates charge is ok.
```
Comment (LauraStelzner-TRI) [inline src/task_behavior_ros/batterystate.py:57]: What does ok mean in this case?  Does it mean currently charging with no failures? Does it mean the charge level is above some threshold? Please clarify

File: src/task_behavior_ros/topic.py [hunk 2]
```diff
@@ -111,19 +111,22 @@ def __init__(self, name, topic_name, topic_type, cb, queue_size=1, *args, **kwar
         self.cb = cb
         self.result = NodeStatus(NodeStatus.ACTIVE)
         self.is_running = False
+        self.latch = latch
```
Comment (LauraStelzner-TRI) [inline src/task_behavior_ros/topic.py:114]: Please add a comment defining latch functionality

File: test/test_batterystate.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.SUCCESS)
+
+        # discharging at max
+        msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_DISCHARGING
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
```
Comment (LauraStelzner-TRI) [inline test/test_batterystate.py:128]: Please add cases for power_supply_status_unknown, not_charging, and full

File: test/test_batterystate.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+class TestChargeOKMonitor(object):
+
+    def setUp(self):
+
```
Comment (LauraStelzner-TRI) [inline test/test_batterystate.py]: extra newline

File: test/test_batterystate.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+from task_behavior_engine.tree import NodeStatus
+from task_behavior_engine.tree import Blackboard
+from task_behavior_ros import batterystate
+
+
+class TestChargeCompleteMonitor(object):
+
+    def setUp(self):
+
```
Comment (LauraStelzner-TRI) [inline test/test_batterystate.py]: extra newline

**Review by LauraStelzner-TRI — CHANGES_REQUESTED** [review]

**Comment by athackst** [comment]
@LauraStelzner-TRI @brandonnorthcutt-tri -- Updated unit tests, please take another look.

**Pull request closed without merging**